internal/dataplane/parser/parser.go

```diff
@@ -182,14 +182,35 @@ func (p *Parser) EnableRegexPathPrefix() {
 	p.flagEnabledRegexPathPrefix = true
 }
 
-func (p *Parser) popTranslationFailures() []TranslationFailure {
-	return p.failuresCollector.PopTranslationFailures()
-}
-
 // -----------------------------------------------------------------------------
 // Parser - Private Methods
 // -----------------------------------------------------------------------------
 
+// registerTranslationFailure should be called when any Kubernetes object translation failure is encountered.
+func (p *Parser) registerTranslationFailure(reason string, causingObjects ...client.Object) {
+	p.failuresCollector.PushTranslationFailure(reason, causingObjects...)
+	p.logTranslationFailure(reason, causingObjects...)
+}
+
+// logTranslationFailure logs an error message signaling that a translation error has occurred along with its reason.
+// `causing_objects` log field is populated with a slice of "GVK, ns/name" strings of translation failure causing objects.
+func (p *Parser) logTranslationFailure(reason string, causingObjects ...client.Object) {
+	objString := func(o client.Object) string {
+		return o.GetObjectKind().GroupVersionKind().String() + ", " + o.GetNamespace() + "/" + o.GetName()
+	}
+
+	objectsStrings := make([]string, 0, len(causingObjects))
+	for _, o := range causingObjects {
+		objectsStrings = append(objectsStrings, objString(o))
+	}
+
+	p.logger.WithField("causing_objects", objectsStrings).Errorf("translation failure has occurred: %s", reason)
+}
```

Comment:
Yeah, I think that makes sense to have a separate log entry for each object, especially since that will work similarly for k8s events. [748bff4](https://github.com/Kong/kubernetes-ingress-controller/pull/3114/commits/748bff4c7443a5f02fe8755a7ca87e58acb5804e)

I stick to the multi-object top-level functions as it does make sense to have the loop hidden, it's always a few lines of code less to write for the caller. Also, the reason for failure is the same for all the objects, so IMO it makes sense to logically aggregate them as a single failure in the collector.